tests/phpunit/unit/MediaWiki/Request/EditRequestSaverTest.php

```diff
@@ -0,0 +1,206 @@
+	}
+
+	public function testEditRequestSavesWith_OtherItems(): void {
+		$firstRequestItem = FluidItem::withId( 'Q1' )->item();
+		$firstRequestOtherItem = FluidItem::withId( 'Q2' )->item();
+		$secondRequestItem = FluidItem::withId( 'Q3' )->item();
+
+		$otherItem = new ReconciliationServiceItem(
+			$firstRequestOtherItem,
+			false
+		);
+
+		$editRequestOne = new EditRequest( $this->reconcileProperty, $firstRequestItem, [ $otherItem ] );
+		$editRequestTwo = new EditRequest( $this->reconcileProperty, $secondRequestItem );
+
+		$itemReconciler = $this->createMock( ItemReconciler::class );
+		$itemReconciler->method( 'reconcileItem' )
+			->withConsecutive(
+				[ $firstRequestItem, $this->reconcileProperty ],
+				[ $secondRequestItem, $this->reconcileProperty ]
+			)
+			->willReturnOnConsecutiveCalls(
+				new ReconciledItem(
+					$firstRequestItem,
+					new ReconciliationServiceItem( $firstRequestItem, false )
+				),
+				new ReconciledItem(
+					$secondRequestItem,
+					new ReconciliationServiceItem( $secondRequestItem, false )
+				)
+			);
+
+		$firstEntityRevision = new EntityRevision( $firstRequestOtherItem, 6543 );
+
+		$firstEditEntity = $this->createMock( EditEntity::class );
+		$firstEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $firstEntityRevision ] ) );
+
+		$secondEntityRevision = new EntityRevision( $firstRequestItem, 1234 );
+
+		$secondEditEntity = $this->createMock( EditEntity::class );
+		$secondEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $secondEntityRevision ] ) );
+
+		$thirdEntityRevision = new EntityRevision( $secondRequestItem, 7891 );
+
+		$thirdEditEntity = $this->createMock( EditEntity::class );
+		$thirdEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $thirdEntityRevision ] ) );
+
+		$editEntityFactory = $this->createMock( MediawikiEditEntityFactory::class );
+		$editEntityFactory
+			->expects( $this->exactly( 3 ) )
+			->method( 'newEditEntity' )
+			->withConsecutive(
+				[ $this->user, $firstRequestOtherItem->getId(), false ],
+				[ $this->user, $firstRequestItem->getId(), false ],
+				[ $this->user, $secondRequestItem->getId(), false ],
+			)
+			->willReturnOnConsecutiveCalls(
+				$firstEditEntity,
+				$secondEditEntity,
+				$thirdEditEntity
+			);
+
+		$saver = $this->newRequestSaver( $editEntityFactory, $itemReconciler );
+
+		// gets updated once saved
+		$this->assertFalse( $otherItem->getRevision() );
+
+		$status = $saver->persistEdits( [
+			$editRequestOne,
+			$editRequestTwo
+		], 'token', $this->user );
+
+		$this->assertTrue( $status->isGood() );
+		$this->assertEquals( $secondEntityRevision, $status->getValue()[0]['revision'] );
+		$this->assertEquals( $thirdEntityRevision, $status->getValue()[1]['revision'] );
+
+		// updated from save
+		$this->assertEquals( $firstEntityRevision->getRevisionId(), $otherItem->getRevision() );
```

Comment:
I think all three `assertEquals()` here could also be `assertSame()`.

Reply:
done